Code/autopkglib/URLDownloader.py

```diff
@@ -125,7 +126,8 @@ def main(self):
         # Download URL.
         url_handle = None
         try:
-            request = urllib2.Request(url=self.env["url"])
+            url = urllib.quote(self.env["url"], 'http://')
```

Comment:
You'd want, at a minimum url = urllib.quote(self.env["url"], ':/') but then it would do the wrong thing if some crazy path had a colon in it.

Again, lets see a real-world example of why it's not acceptable to require that the URL provided to URLDownloder be properly encoded from the start.